base/array.jl

```diff
@@ -791,9 +791,10 @@ function find(testf::Function, A::AbstractArray)
     I
 end
 
-function find(A::AbstractArray)
-    nnzA = countnz(A)
-    I = similar(A, Int, nnzA)
+function find(A)
+    cA = typeof(A) <: AbstractArray ? A : collect(A)
```

Comment:
Yeah, I had conditioned on the type to avoid unnecessarily `collect`ing. (Side note: in your example, `UnitRange <: AbstractArray`, so it wouldn't get `collect`ed. But I get what you mean.) Great idea regarding `Vector`, I'll do that. Thanks so much for the feedback! 😄